source/codegen/common_helpers.py

```diff
@@ -13,24 +13,36 @@ def is_enum(parameter):
 def is_struct(parameter):
   return parameter["type"].startswith("struct")
 
+def get_input_and_output_custom_types(functions):
+  '''Returns a set of custom types used by input and output parameters separately.'''
+  input_custom_types = set()
+  output_custom_types = set()
+  for function in functions:
+    for parameter in functions[function]["parameters"]:
+        if is_struct(parameter) and is_input_parameter(parameter) and is_array(parameter["type"]):
+            input_custom_types.add(get_underlying_type_name(parameter["type"]))
+        if is_struct(parameter) and is_output_parameter(parameter) and is_array(parameter["type"]):
```

Comment:
This can be `elif`, since if it is `input_param` it can't be an `output_param` as well.

Reply:
Changed it to elif